src/server/pfs/drive/driver.go

```diff
@@ -970,14 +973,16 @@ func deleteFromDir(diffInfo *pfs.DiffInfo, child *pfs.File) {
 }
 
 type fileReader struct {
-	blockClient pfsclient.BlockAPIClient
-	blockRefs   []*pfs.BlockRef
-	index       int
-	reader      io.Reader
-	offset      int64
-	size        int64
-	ctx         context.Context
-	cancel      context.CancelFunc
+	blockClient               pfsclient.BlockAPIClient
+	blockRefs                 []*pfs.BlockRef
+	index                     int
+	reader                    io.Reader
+	offset                    int64
+	size                      int64
+	bytesReadFromCurrentBlock uint64
+	currentBlockSize          uint64
+	ctx                       context.Context
+	cancel                    context.CancelFunc
```

Comment:
These fields introduce a number of ways for the data structure to become inconsistent which will likely lead to bugs in the future. `currentBlockSize` is only ever set to `pfsserver.ByteRangeSize(r.blockRef().Range)` so it should just be a method that returns that. Otherwise you're implicitly adding the constraint that every time someone updates `index` they have to update this field as well. Eventually someone's going to forget to do that and it'll turn into a really tricky bug.